### PR TITLE
GTEST/UCT/SOCKADDR: port sockaddr tests to new API

### PR DESCRIPTION
**Pull request by evgeny-leksikov**
## What
 - port all sockaddr tests to new API
 - add basic sockaddr test with legacy API

## Why ?
test coverage

### CODE REVIEW

**Comment by evgeny-leksikov** [comment]
bot:pipe:retest

**Comment by evgeny-leksikov** [comment]
timed out
bot:pipe:retest

File: test/gtest/uct/ib/test_sockaddr.cc [hunk 1]
```diff
@@ -1149,3 +1181,105 @@ UCS_TEST_P(test_uct_sockaddr_multiple_cms, server_switch_cm)
 }
 
 UCT_INSTANTIATE_SOCKADDR_TEST_CASE(test_uct_sockaddr_multiple_cms)
+
+class test_uct_sockaddr_legacy : public test_uct_sockaddr
```
Comment (alinask) [inline test/gtest/uct/ib/test_sockaddr.cc:1185]: this class is to test the previous flow only?
why do we need it if the new code covers both cases.

File: src/uct/ib/rdmacm/rdmacm_cm_ep.c [hunk 2]
```diff
@@ -236,7 +236,7 @@ ucs_status_t uct_rdmacm_cm_ep_resolve_cb(uct_rdmacm_cm_ep_t *cep)
 
     args.field_mask = UCT_CM_EP_RESOLVE_ARGS_FIELD_DEV_NAME;
     uct_rdmacm_cm_id_to_dev_name(cep->id, args.dev_name);
-    return uct_cm_ep_resolve_cb(cep->super.user_data, &args);
+    return uct_cm_ep_resolve_cb(&cep->super, &args);
```
Comment (alinask) [inline src/uct/ib/rdmacm/rdmacm_cm_ep.c:239]: this was already merged in another pr right?

File: test/gtest/uct/ib/test_sockaddr.cc (hunk 1 above)
Comment (evgeny-leksikov) [reply test/gtest/uct/ib/test_sockaddr.cc:1185]: > this class is to test the previous flow only?

yes
> why do we need it if the new code covers both cases.

this is implementation details, need to cover API as well

File: src/uct/ib/rdmacm/rdmacm_cm_ep.c (hunk 2 above)
Comment (evgeny-leksikov) [reply src/uct/ib/rdmacm/rdmacm_cm_ep.c:239]: yes, it was cherry-picked to pass CI, the commit should be dropped on merge automatically or on squash after review if needed

**Review by alinask — APPROVED** [review]

**Comment by alinask** [comment]
@yosefe Can you please review?

File: test/gtest/uct/uct_test.h [hunk 3]
```diff
@@ -130,6 +130,18 @@ class uct_test : public testing::TestWithParam<const resource*>,
...
+        struct sockaddr_user_data {
+            sockaddr_user_data(uct_test *test, entity *entity,
+                               unsigned ep_index) :
+                    m_test(test), m_entity(entity), m_ep_index(ep_index)
+            {
+            }
+
+            uct_test *m_test;
+            entity   *m_entity;
+            unsigned m_ep_index;
```
Comment (yosefe) [inline test/gtest/uct/uct_test.h:142]: i'd remove m_ prefix
since this is more a POD than a class

File: test/gtest/uct/ib/test_sockaddr.cc (hunk 1 above)
Comment (yosefe) [reply test/gtest/uct/ib/test_sockaddr.cc:1185]: pls add comment for this test class

File: test/gtest/uct/uct_test.h [hunk 4]
```diff
@@ -133,13 +133,13 @@ class uct_test : public testing::TestWithParam<const resource*>,
...
-                    m_test(test), m_entity(entity), m_ep_index(ep_index)
+                    test_p(test), entity_p(entity), entity_ep_index(ep_index)
             {
             }
 
-            uct_test *m_test;
-            entity   *m_entity;
-            unsigned m_ep_index;
+            uct_test *test_p;
+            entity   *entity_p;
```
Comment (yosefe) [inline test/gtest/uct/uct_test.h:141]: minor: i'd remove _p suffix for consistency (we use _p for pointer-to-handle func params in C code)

Comment (evgeny-leksikov) [reply test/gtest/uct/uct_test.h:141]: there are 2 issues :)
1) need new names for args in ctor
2) compiler warns about `entity *entity` definition

maybe better to revert `m_*` prefixes?

**Review by yosefe — APPROVED** [review]

**Comment by evgeny-leksikov** [comment]
@yosefe ok to merge?

**Pull request merged**